tenpy/networks/mps.py

```diff
@@ -4403,6 +4404,92 @@ def add(self, other, alpha, beta, cutoff=1.e-15):
         psi.canonical_form_finite(renormalize=False, cutoff=cutoff)
         return psi
 
+    def extend(self, others=[], trunc_par={'svd_min': 1.e-8}, do_canonicalize=False):
+        """Return an MPS (in place) which represents the exact same physical state as ``|self>`` but has 
+        the basis extended by additional MPS. We follow the approach 
+        from https://arxiv.org/abs/2005.06104.
+
+        Works (for now) only for 'finite' boundary conditions.
+        While in principle this could work for segments, ._gauge_compatible_vL_vR doesn't work
+        for 'segment' boundary conditions. In principle, this could be implemented.
+
+        If we pass in no new MPSs, we simply add random orthogonal states.
+
+        Parameters
+        ----------
+        others : list of :class:`MPS` or an empty list ([])
+            MPS of same length used to extend the basis of self
+            If None, we add random basis states.
+        trunc_par : dict
+            Parameters for truncation, see :cfg:config:`truncation`.
+        do_canonicalize : Boolean
+            Do we convert to canonical form at the end? This has the tendency to remove some trivial
+            singular values which negates the purpose of this function.
```

Comment:
The reason why I left it is that if one wants to have a "sane" MPS in memory after this function (i.e. one where the SVs on each bond have the proper length, where `psi.chi` actually returns the correct bond dimension, and where `psi.norm_test()` is all zeros), one should call canonical form. When using Krylov vectors to expand the basis, I am finding that canonicalizing the new MPS doesn't remove all of the new SVs that were added since the cutoff is 0.

Reply:
What I can do is remove this option but clearly document to the user that the MPS is not canonical and ought to be used in a L->R->L sweep (i.e. DMRG or TDVP), where the sweeping will fix the MPS structure.

Reply:
And your comment above about expanding the SVs on the bonds with zeros will ensure that `psi.chi` returns sensible values.